src/main/java/de/tum/cit/aet/artemis/communication/web/FaqResource.java

```diff
@@ -105,9 +107,13 @@ public ResponseEntity<FaqDTO> updateFaq(@RequestBody Faq faq, @PathVariable Long
         if (faqId == null || !faqId.equals(faq.getId())) {
             throw new BadRequestAlertException("Id of FAQ and path must match", ENTITY_NAME, "idNull");
         }
-        checkPriviledgeForAcceptedElseThrow(faq, courseId);
+        if (faq.getFaqState() == FaqState.ACCEPTED) {
+            checkRoleForCourse(courseId, Role.INSTRUCTOR);
+        }
         Faq existingFaq = faqRepository.findByIdElseThrow(faqId);
-        checkPriviledgeForAcceptedElseThrow(existingFaq, courseId);
+        if (existingFaq.getFaqState() == FaqState.ACCEPTED) {
+            checkRoleForCourse(courseId, Role.INSTRUCTOR);
+        }
```

Comment:
We have this check 3 times in this file, I think it was good style to have it wrapped in a method.
If the role would be changed that is allowed to accept the FAQ this would now need to be changed in 3 places instead of 1. 

Could we extract the method here again? 😅